noria-server/dataflow/src/state/persistent_state.rs

```diff
@@ -206,31 +128,32 @@ impl State for PersistentState {
         // We'll store all the pointers (or values if this is index 0) for
         // this index in its own column family:
         let index_id = self.indices.len().to_string();
-        let column_family = self
-            .db
-            .as_mut()
-            .unwrap()
-            .create_cf(&index_id, &self.db_opts)
-            .unwrap();
+        let db = self.db.as_mut().unwrap();
+        db.create_cf(&index_id, &self.db_opts).unwrap();
 
         // Build the new index for existing values:
         if !self.indices.is_empty() {
-            for chunk in self.all_rows().chunks(INDEX_BATCH_SIZE).into_iter() {
```

Comment:
Not a huge deal, but I couldn't get my `all_rows()` helper to work here: 

```rust
error[E0502]: cannot borrow `*self` as immutable because it is also borrowed as mutable
   --> noria-server/dataflow/src/state/persistent_state.rs:139:26
    |
131 |         let column_family = self.db
    |                             ------- mutable borrow occurs here
...
139 |             for chunk in self.all_rows().chunks(INDEX_BATCH_SIZE).into_iter() {
    |                          ^^^^ immutable borrow occurs here
...
145 |                     batch.put_cf(column_family, &key, value).unwrap();
    |                                  ------------- mutable borrow later used here

```

Reply:
Ah, this is an artifact of you now setting
```rust
let db = self.db.as_mut().unwrap();
```
And then using `db` repeatedly below, instead of using `self.db.get_mut()` repeatedly below. When you call `self`, the compiler detects that `self` is already borrowed in `db`, and so it disallows the call.